Spigot-Server-Patches/0010-Adventure.patch

```diff
@@ -1862,6 +1862,26 @@ index ae735836accc6bf6f0831f72ff882720b69df792..d3ae5cadd88f9012203d2c04cbe38af9
      @Override
      public String getCustomName() {
          TileEntityBeacon beacon = this.getSnapshot();
+diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftCommandBlock.java b/src/main/java/org/bukkit/craftbukkit/block/CraftCommandBlock.java
+index a99d79153ada6bf2aa6a8c357b915608b6771a70..93cc743b4dca74ff6417c014985d3c983c2ab242 100644
+--- a/src/main/java/org/bukkit/craftbukkit/block/CraftCommandBlock.java
++++ b/src/main/java/org/bukkit/craftbukkit/block/CraftCommandBlock.java
+@@ -35,4 +35,15 @@ public class CraftCommandBlock extends CraftBlockEntityState<TileEntityCommand>
+     public void setName(String name) {
+         getSnapshot().getCommandBlock().setName(CraftChatMessage.fromStringOrNull(name != null ? name : "@"));
+     }
++    // Paper start
++    @Override
++    public void name(net.kyori.adventure.text.Component name) {
++        getSnapshot().getCommandBlock().setName(name == null ? new net.minecraft.network.chat.ChatComponentText("@") : io.papermc.paper.adventure.PaperAdventure.asVanilla(name));
```

Comment:
Does this actually work? It's been a while since I touched any of this, but aren't you just updating the snapshot and not the real thing?

Reply:
Yes. The plugin must call BlockState#update themselves to make the change actually happen.